### PR TITLE
fix(mypy): fix mypy errors for src/sentry/api/serializers/models (part 1)

### PR DESCRIPTION
**Pull request by hubertsentry**
Fix mypy errors under src/sentry/api/serializers/models

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: src/sentry/api/serializers/models/release_file.py [hunk 1]
```diff
@@ -29,8 +30,8 @@ def decode_release_file_id(id: str):
     except ValueError:
         decoded = urlsafe_b64decode(id).decode()
         dist, url = decoded.split("_", 1)
-        dist = dist or None
-        return dist, url
+        ret_dist: Optional[str] = dist or None
+        return ret_dist, url
```
Comment (asottile-sentry) [inline src/sentry/api/serializers/models/release_file.py:34]: probably don't need the annotation now that it's renamed -- can probably also be `return dist or None, url` without needing a separate variable

File: src/sentry/api/serializers/models/tagvalue.py [hunk 2]
```diff
@@ -21,6 +23,7 @@ def get_attrs(self, item_list, user):
         return result
 
     def serialize(self, obj, attrs, user):
+        result: Dict[str, Any]
```
Comment (asottile-sentry) [inline src/sentry/api/serializers/models/tagvalue.py:26]: I usually annotate the first assignment instead of above them all

**Review by asottile-sentry — COMMENTED** [review]
nice!

**Review by asottile-sentry — APPROVED** [review]

**Review by asottile-sentry — APPROVED** [review]

**Comment by hubertsentry** [comment]
Sorry, it took longer then I thought. It seems like one of my change caused [openapi-diff issue](https://github.com/getsentry/sentry/actions/runs/5746722279/job/15576639436). [My change](https://github.com/getsentry/sentry/pull/53905/commits/d5e146251dc93dc131b2d6492ac07725cc19fd7d) is probably doing the right thing, however, it will make api schema incompatible. Let me revert the change for that file now, I will see how I can address the issue later.

**Pull request merged**